extensions/kinesis/src/main/java/com/hazelcast/jet/kinesis/impl/sink/ShardCountMonitorImpl.java

```diff
@@ -42,35 +48,27 @@
      */
     private static final double RATIO_OF_DESCRIBE_STREAM_RATE_UTILIZED = 0.1;
 
-    private final AtomicInteger shardCount;
+    private final AtomicInteger shardCount = new AtomicInteger();
     private final RandomizedRateTracker describeStreamRateTracker;
     private final RetryTracker describeStreamRetryTracker;
 
     private Future<DescribeStreamSummaryResult> describeStreamResult;
     private long nextDescribeStreamTime;
 
-    public ShardCountMonitor(
-            int totalInstances,
+    ShardCountMonitorImpl(
+            int knownShardCountInstances,
```

Comment:
I find this name misleading. It suggests that it's an integer related to the count of shards, as opposed to the count of monitors. Morever, it sounds a bit like an uncertain value, that there is an unknown part of the count and this is like an estimate. How about using `monitorInstances` or keeping `totalInstances` instead?

Reply:
I renamed to `knownShardCounterInstances`.

>it sounds a bit like an uncertain value
That was exactly the point. There can be others calling shard count, this value includes the number of counters we know about. And that's why we run at 10% or the limit...